google-cloud-storage/src/main/scala/akka/stream/alpakka/googlecloud/storage/FailedUpload.scala

```diff
@@ -0,0 +1,9 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.googlecloud.storage
+
+import scala.collection.immutable.Seq
+
+final case class FailedUpload(reasons: Seq[Throwable]) extends Exception(reasons.map(_.getMessage).mkString(", "))
```

Comment:
This should be converted to non-case class and introduced a Java API getter for reasons (as other model classes have).